### PR TITLE
Add accessibility attributes to the preferences dialog

### PR DESCRIPTION
**Pull request by iainx**
This adds accessibility attributes to the preferences panels and their dialogs, except for the code formatting panes and the F# settings.

Help descriptions are missing from some panes that are only checkboxes, but the checkbox's labels are usually descriptive enough to not need them.

### CODE REVIEW

**Comment by iainx** [comment]
This fixes 53448

**Comment by iainx** [comment]
☕️ for @Therzok  😸

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.OptionPanels/GlobalAuthorInformationPanel.cs [hunk 1]
```diff
@@ -61,8 +62,33 @@ public GlobalAuthorInformationPanelWidget()
...
 			trademarkEntry.Text = AuthorInformation.Default.Trademark ?? "";
+
+			SetupAccessibility ();
 		}
-		
+
+		void SetupAccessibility ()
+		{
+			nameEntry.SetCommonAccessibilityAttributes ("AuthorInformationPanel.nameEntry", "",
+														GettextCatalog.GetString ("Enter the author name"));
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.OptionPanels/GlobalAuthorInformationPanel.cs]: `author's name`?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.OptionPanels/AuthorInformationPanel.cs [hunk 2]
```diff
@@ -74,8 +76,31 @@ public AuthorInformationPanelWidget (AuthorInformation info)
...
 			UseDefaultToggled (this, EventArgs.Empty);
+
+			SetupAccessibility ();
 		}
-		
+
+		void SetupAccessibility ()
+		{
+			nameEntry.SetCommonAccessibilityAttributes ("AuthorInformationPanel.nameEntry", "",
+			                                            GettextCatalog.GetString ("Enter the author name"));
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.OptionPanels/AuthorInformationPanel.cs]: `author's name`?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.ExternalTools/ExternalToolPanel.cs [hunk 3]
```diff
@@ -135,8 +136,47 @@ public ExternalToolPanelWidget ()
...
 			SelectionChanged (this, EventArgs.Empty);
+
+			SetupAccessibility ();
 		}
-		
+
+		void SetupAccessibility ()
+		{
+			addButton.SetCommonAccessibilityAttributes ("ExternalTools.Add", null,
+			                                            GettextCatalog.GetString ("Click to add a new external tool"));
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.ExternalTools/ExternalToolPanel.cs]: general inconsistency:
`this command` and `this external command`

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.OptionPanels/IDEStyleOptionsPanel.cs [hunk 4]
```diff
@@ -88,6 +88,14 @@ public IDEStyleOptionsPanelWidget ()
 			this.Build();
 			Load ();
 
+			comboTheme.SetCommonAccessibilityAttributes ("IDEStyleOptionsPanel.Theme", null,
+			                                             GettextCatalog.GetString ("Select the user interface theme"));
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.OptionPanels/IDEStyleOptionsPanel.cs]: 👏 syntax highlighting from github

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.OptionPanels/TasksOptionsPanel.cs [hunk 5]
```diff
@@ -69,6 +70,41 @@ public TasksPanelWidget ()
 			entryToken.Changed += new EventHandler (Validate);
 
 			Styles.Changed += HandleUserInterfaceThemeChanged;
+			SetupAccessibility ();
+		}
+
+		void SetupAccessibility ()
+		{
+			comboPriority.SetCommonAccessibilityAttributes ("TasksPanel.priorityCombo", null,
+			                                                GettextCatalog.GetString ("Select the priority for this token"));
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.OptionPanels/TasksOptionsPanel.cs]: inconsistency `this` and `selected`

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.OptionPanels/AssemblyFoldersPanel.cs [hunk 6]
```diff
@@ -55,6 +56,16 @@ public AssemblyFoldersPanelWidget ()
...
 			label1.LabelProp = MonoDevelop.Core.BrandingService.BrandApplicationName (label1.LabelProp);
 			selector.Directories = new List<string> (Runtime.SystemAssemblyService.UserAssemblyContext.Directories);
+
+			SetupAccessibility ();
+		}
+
+		void SetupAccessibility ()
+		{
+			selector.EntryAccessible.SetCommonAttributes (null, null,
+			                                              GettextCatalog.GetString ("Enter a folder to search for assemblies and packages"));
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.OptionPanels/AssemblyFoldersPanel.cs]: `to search in for`

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebuggerOptionsPanelWidget.cs [hunk 7]
```diff
@@ -99,6 +102,30 @@ void Build ()
...
+			checkAllowToString.SetCommonAccessibilityAttributes ("DebuggerPanel.allowToString", "",
+			                                                     GettextCatalog.GetString ("Check to call string-conversion functions on objects in the Variables windows"));
+			checkShowBaseGroup.SetCommonAccessibilityAttributes ("DebuggerPanel.showBaseGroup", "",
+			                                                     GettextCatalog.GetString ("Check to show inherited class members in a base class group"));
+			checkGroupPrivate.SetCommonAccessibilityAttributes ("DebuggerPanel.groupPrivate", "",
+			                                                    GettextCatalog.GetString ("Check to group non-public members in the Variables windows"));
+			checkGroupStatic.SetCommonAccessibilityAttributes ("DebuggerPanel.groupStatic", "",
+			                                                   GettextCatalog.GetString ("Check to group static members in the Variables windows"));
+			spinTimeout.SetCommonAccessibilityAttributes ("DebuggerPanel.timeout", evalLabel,
+			                                              GettextCatalog.GetString ("Set the length of time the evaluation will wait before giving up"));
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebuggerOptionsPanelWidget.cs]: `Set the timeout length for evaluation`?

Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebuggerOptionsPanelWidget.cs]: Maybe also mention it's in millis?

File: main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.OptionPanels/GeneralOptionsPanel.cs [hunk 8]
```diff
@@ -42,6 +43,20 @@ public GeneralOptionsPanel()
...
+
+			SetupAccessibility ();
+		}
+
+		void SetupAccessibility ()
+		{
+			comboboxLineEndings.SetCommonAccessibilityAttributes ("SourceEditorGeneral.lineEndings", label1,
+			                                                      GettextCatalog.GetString ("Select how to handle line ending conversions"));
+			foldingCheckbutton.SetCommonAccessibilityAttributes ("SourceEditorGeneral.folding", "",
+			                                                     GettextCatalog.GetString ("Check to enable line folding"));
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.OptionPanels/GeneralOptionsPanel.cs]: `code folding` https://en.wikipedia.org/wiki/Code_folding

File: main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.OptionPanels/MarkerPanel.cs [hunk 9]
```diff
@@ -69,6 +70,37 @@ public MarkerPanel()
...
+			checkbuttonSpaces.SetCommonAccessibilityAttributes ("MarkerPanel.spaces", "",
+			                                                    GettextCatalog.GetString ("Check to show spaces when showing invisible characters"));
+			checkbuttonLineEndings.SetCommonAccessibilityAttributes ("MarkerPanel.lineEndings", "",
+			                                                         GettextCatalog.GetString ("Check to show line endings when showing invisible characters"));
+			showRulerCheckbutton.SetCommonAccessibilityAttributes ("MarkerPanel.showRuler", "",
+			                                                       GettextCatalog.GetString ("Check to show the column ruler"));
+			showLineNumbersCheckbutton.SetCommonAccessibilityAttributes ("MarkerPanel.showLineNumbers", "",
+			                                                             GettextCatalog.GetString ("Check to show line numbers"));
+			highlightCurrentLineCheckbutton.SetCommonAccessibilityAttributes ("MarkerPanel.highlightCurrentLine", "",
+			                                                                  GettextCatalog.GetString ("Check to highlight to current line"));
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.OptionPanels/MarkerPanel.cs]: `the current line`

File: main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.OptionPanels/MarkerPanel.cs [hunk 10]
```diff
@@ -69,6 +70,37 @@ public MarkerPanel()
...
+			highlightMatchingBracketCheckbutton.SetCommonAccessibilityAttributes ("MarkerPanel.highlightMatching", "",
+			                                                                      GettextCatalog.GetString ("Check to highlight the matching bracket"));
+			enableHighlightUsagesCheckbutton.SetCommonAccessibilityAttributes ("MarkerPanel.highlightUsages", "",
+			                                                                   GettextCatalog.GetString ("Check to highlight identifier references"));
+			enableAnimationCheckbutton1.SetCommonAccessibilityAttributes ("MarkerPanel.enableAnim", "",
+			                                                              GettextCatalog.GetString ("Check to enable animations in the text editor"));
+			enableQuickDiffCheckbutton.SetCommonAccessibilityAttributes ("MarkerPanel.quickDiff", "",
+			                                                             GettextCatalog.GetString ("Check to highlight changed line"));
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor.OptionPanels/MarkerPanel.cs]: `the changed line`

**Comment by Therzok** [comment]
Code-wise, it looks ok to me.

**Comment by slluis** [comment]
I'll merge as is. Remaining comments can be fixed in another PR, since this is in Iain's fork and he is going to be away for a while.

**Pull request merged**